wcfsetup/install/files/js/WCF.js

```diff
@@ -5966,39 +5966,46 @@ if (COMPILER_TARGET_DEFAULT) {
 		/**
 		 * Callback for file uploads.
 		 *
-		 * @param        object                event
+		 * @param        object              event
 		 * @param        File                file
 		 * @param        Blob                blob
-		 * @return        integer
+		 * @param        Array|FileList      list of files
+		 * @return       integer
 		 */
-		_upload: function (event, file, blob) {
+		_upload: function (event, file, blob, files) {
 			var $uploadID = null;
 			var $files = [];
-			if (file) {
-				$files.push(file);
-			}
-			else if (blob) {
-				var $ext = '';
-				switch (blob.type) {
-					case 'image/png':
-						$ext = '.png';
-						break;
-					
-					case 'image/jpeg':
-						$ext = '.jpg';
-						break;
-					
-					case 'image/gif':
-						$ext = '.gif';
-						break;
-				}
-				
-				$files.push({
-					name: 'pasted-from-clipboard' + $ext
-				});
+			
+			if (typeof files !== 'undefined') {
+				$files = files;
 			}
 			else {
-				$files = this._fileUpload.prop('files');
+				if (file) {
+					$files.push(file);
+				}
+				else if (blob) {
+					var $ext = '';
+					switch (blob.type) {
+						case 'image/png':
+							$ext = '.png';
```

Comment:
That's a bit counter-intuitive, we usually only store the actual extension in variables, not the dot itself.

Reply:
That code has been left untouched :)